test/e2e/jobs/jobs.test.ts

```diff
@@ -0,0 +1,154 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { GenericContainer, Network, StartedNetwork, StartedTestContainer, TestContainers, Wait } from "testcontainers";
+// import { LogWaitStrategy } from "testcontaineclienrs/build/wait-strategies/log-wait-strategy";
+import { CommunicationProtocolEnum, DaprClient, DaprServer } from "../../../src";
+// import { AbstractWaitStrategy } from "testcontainers/build/wait-strategies/wait-strategy";
+
+jest.setTimeout(40 * 1000);
+
+describe("Jobs End to End", () => {
+
+    let network: StartedNetwork | null = null;
+    let daprScheduler: StartedTestContainer | null = null;
+    let daprd: StartedTestContainer | null = null;
+    let server: DaprServer | null = null;
+    let client: DaprClient | null = null;
+
+    beforeAll(async () => {
+
+      await TestContainers.exposeHostPorts(8070);
+
+      network = await new Network().start();
+
+      daprScheduler = await (new GenericContainer("ghcr.io/dapr/dapr")
+        .withName("dapr-js-sdk-test-scheduler")
+        .withNetwork(network)
+        .withNetworkAliases("scheduler")
+        .withExposedPorts(8083)
+        .withCommand([
+          "./scheduler",
+          // note: Don't think this is necessary, buuuuut????
+          "--listen-address", "0.0.0.0",
+          "--port", "8083",
+          "--log-level", "debug",
+          // note: This feels redundant, but here as yet another thing I've tried.
+          // "--mode", "standalone",
+        ])
+        .withTmpFs({
+          "/data": "rw",
+        })
+        // note: Because dapr containers don't have `sh` or `bash` inside, this is kind of the best health check.
+        .withWaitStrategy(Wait.forLogMessage("api is ready").withStartupTimeout(10000))
+        .start());
+
+        daprd = await (new GenericContainer("ghcr.io/dapr/dapr")
+          .withName("dapr-js-sdk-test-daemon")
+          .withNetwork(network)
+          .withNetworkAliases("daprd")
+          .withExposedPorts(8081, 8082)
+          .withCommand([
+            "./daprd",
+            "--app-id", "dapr-js-sdk-testing",
+            "--app-channel-address", "host.testcontainers.internal",
+            "--app-port", "8070",
+            "--dapr-grpc-port", "8081",
+            "--dapr-http-port", "8082",
+            "--scheduler-host-address", `scheduler:8083`,
+            "--placement-host-address", "",
+            "--enable-metrics", "false",
+            "--log-level", "debug",
+          ])
+          // note: Because dapr containers don't have `sh` or `bash` inside, this is kind of the best health check.
+          .withWaitStrategy(Wait.forLogMessage("HTTP server is running on port").withStartupTimeout(10000))
+          .start());
+
+      console.info(`Scheduler: ${getIp(daprScheduler)}`);
+      console.info(`Daemon: ${getIp(daprd)}`);
+
+      server = new DaprServer({
```

Comment:
Do you need DaprServer or can you just use an express server?

I think there's a chicken-egg problem with DaprServer in these tests because exposeHostPorts should be called after the server starts but before starting daprd.